CONTRIBUTING.rst

```diff
@@ -0,0 +1,480 @@
+when you open a pull request will tell you if you've introduced any new
+lines that aren't covered by a test.
+
+You can also run unit tests locally, and you should do this periodically
+to test your code. To do this, you will need to install the optional
+dependencies needed for running tests.
+
+::
+
+   >> pip install -e ".[test]"
+
+This will install the optional 'test' dependencies specified in
+``pyproject.toml`` that don't install by default. The package ``pytest``
+is one of these and is what's used to run the tests. ``pytest`` searches
+through all the directories in your repository (underneath the directory
+from which it was invoked command line) and looks for any directories
+called 'test' or .py files with the word 'test' in the name. Functions
+in these files will be executed.
+
+To run all of the ``drizzlepac`` unit tests, while in the
+``drizzlepac/`` level directory, simply run the command:
+
+::
+
+   >> pytest
+
+If you want to run all the tests for a single instrument/directory, for example
+``WFC3``, you can run this from 'drizzlepac/wfc3'.
+
+To run all tests within a single test file (for example, all tests in
+``test_wfc3``). 
+
+::
+   
+   >> pytest drizzlepac/wfc3/test_wfc3.py
+
+Configuring pytest for unit tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``pytest`` can be configured with many different flags - see the `pytest
+documentation <https://docs.pytest.org/en/7.1.x/contents.html>`__ to see
+all of these. Here we will summarize a few of the most useful options.
+
+If you are writing and debugging a test, you may find it helpful to have
+print statements printed to the terminal while running tests (which
+doesn't happen by default). To do this,
+
+::
+
+   >> pytest -s
+
+If you want to only run a specific test function within a file (or only
+tests with names that contain a certain string):
+
+::
+
+   >> pytest -k testname.
+
+This will search all files under the directory you're in for files or
+functions with the string 'test' in them, and within those files run
+only the test functions that contain the string ``testname``.
+
+Within the test files themselves, decorators can be used to control the
+behavior of the test. Some of the more useful decorators include: 1.
+@pytest.mark.parametrize can be used to run a single test on multiples
+sets of input parameters 2. @pytest.skip can be used to skip tests
+altogether, or under specific conditions (for example, only when being
+run by the CI) 3. @pytest.fixture to declare a
+`fixture <https://docs.pytest.org/en/7.1.x/explanation/fixtures.html>`__
+1. @pytest.mark.xfail will make a test pass only if it fails.
+
+Simultaneously developing ``drizzlepac`` and one of its dependencies
+--------------------------------------------------------------------
+
+If you encounter the scenario where you wish to simultaneously make
+changes in ``drizzlepac`` and also in one of its dependencies like
+``stcal`` or ``stpipe``, we recommend that you create a new environment
```

Comment:
'stcal' and 'stpipe' are not dependencies of 'drizzlepac'.  Pick something for illustration from pyproject.toml.